js/EmbedRequest.js

```diff
@@ -0,0 +1,126 @@
+      "type": "array",
+      "description": "The ancestors of the topic, in order, from the parent to the root",
+      "items": {
+        "$ref": "#/definitions/topic"
+      }
+    },
+    "language": {
+      "type": "string",
+      "description": "Language code from https://github.com/learningequality/le-utils/blob/main/le_utils/resources/languagelookup.json",
+      "pattern": "^[a-z]{2,3}(?:-[a-zA-Z]+)?$"
+    },
+    "topic": {
+      "type": "object",
+      "description": "A topic in the tree structure",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the topic content node on Studio"
+        },
+        "title": {
+          "type": "string",
+          "description": "The title of the topic"
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the topic"
+        },
+        "language": {
+          "$ref": "#/definitions/language"
+        },
+        "ancestors": {
+          "$ref": "#/definitions/ancestors"
+        }
+      },
+      "required": [
+        "id",
+        "title",
+        "description"
+      ]
+    },
+    "resource": {
+      "type": "object",
+      "description": "The key textual metadata and data for a content resource",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the content resource"
+        },
+        "title": {
+          "type": "string",
+          "description": "The title of the content resource"
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the content resource"
+        },
+        "text": {
+          "type": "string",
+          "description": "The cleaned up text extracted from the content resource (in markdown or plaintext format)"
+        },
+        "language": {
+          "$ref": "#/definitions/language"
+        }
+      },
+      "required": [
+        "id",
+        "title",
+        "description",
+        "text"
+      ]
+    }
+  },
+  "properties": {
+    "topics": {
+      "type": "array",
+      "description": "A list of topics to embed",
+      "items": {
+        "$ref": "#/definitions/topic"
+      }
+    },
+    "resources": {
+      "type": "array",
+      "description": "A list of content resources to embed",
+      "items": {
+        "$ref": "#/definitions/resource"
+      }
+    },
+    "metadata": {
+      "type": "object",
+      "description": "The metadata of the channel for logging purposes",
+      "properties": {
+        "channel_id": {
+          "type": "string",
+          "description": "The ID of the channel"
+        },
+        "channel_title": {
+          "type": "string",
+          "description": "The title of the channel"
+        }
+      },
+      "required": [
+        "channel_id",
+        "channel_title"
+      ]
```

Comment:
I'd recommend not having these be required, since this API may be called outside of the Studio context (where we don't have channel metadata).

Reply:
Based on this, Its probably safe to remove the 2 altogether as it no longer makes sense under the circumstances.